lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py

```diff
@@ -124,6 +124,33 @@ def test___repr__(self):
         )
         self.assertEqual(expected, self.mesh.__repr__())
 
+    def test___eq__(self):
+        # The dimension names do not participate in equality.
+        equivalent_kwargs = self.kwargs.copy()
+        equivalent_kwargs["node_dimension"] = "something_else"
+        equivalent = mesh.Mesh(**equivalent_kwargs)
+        self.assertEqual(equivalent, self.mesh)
+
+    def test_different(self):
+        # 2021-11-26 currently not possible to have a metadata-only difference
+        #  - only topology_dimension makes a difference and that also mandates
+        #  different connectivities.
```

Comment:
I'm not sure what you mean by this. Could you elaborate?

Reply:
I would ideally have also tested that a difference would be detected if ONLY the metadata were different and the `Coord`s and `Connectivity`s were identical. But this isn't possible since the only possible metadata difference is in `topology_dimension`, and different `topology_dimension`s require different `Connectivity`s.

Does that help? Do you have any ideas for a phrasing that's easier to understand?

Reply:
So do you mean the metadata of the Mesh itself? Because I imagine testing for a different `start_index` for example could also be worthwhile, which is in the `Connectivity` metadata.

Reply:
>So do you mean the metadata of the Mesh itself?

Yes I specifically mean the contents of the `MeshMetadata` instance attached to the `Mesh`.

---

>Because I imagine testing for a different `start_index` for example could also be worthwhile, which is in the `Connectivity` metadata.

That certainly would qualify as a difference, but such a check is deferred to `Connectivity.__eq__()` so from the point of view of this test it's no different from the existing lines:

https://github.com/SciTools/iris/blob/3d0d0aa00c9f2ac6723dcbd8fab3d57597fb25f2/lib/iris/tests/unit/experimental/ugrid/mesh/test_Mesh.py#L147-L152

Reply:
That makes sense. In which case I think it's worth expanding the test in the connectivity equality check that makes sure it is able to handle metadata in equality properly. https://github.com/SciTools/iris/blob/b03aa7c16cba7e4947c93301e99494736c0bd977/lib/iris/tests/unit/experimental/ugrid/mesh/test_Connectivity.py#L129-L133

Reply:
Found a way to have a metadata difference: e5b1d2385c889acd8151aca808fcf1c895f32fdf

We have agreed that the missing `Connectivity` test coverage mentioned above is beyond the scope of this PR.